src/popup/components/TransactionList.vue

```diff
@@ -50,163 +40,189 @@ import {
   ref,
   watch,
 } from '@vue/composition-api';
-import type {
-  INetwork,
-  ITokenList,
-  ITx,
-} from '../../types';
 import {
   TXS_PER_PAGE,
+  getTransaction,
+  getMultisigTransaction,
+  getInnerTransaction,
   AETERNITY_CONTRACT_ID,
-  MOBILE_WIDTH,
+  isTxDex,
   TX_DIRECTION,
-  defaultTransactionSortingCallback,
-  getInnerTransaction,
+  sortTransactionByDate,
+  pipe,
 } from '../utils';
-import { useGetter, useState } from '../../composables/vuex';
+import { useDispatch, useGetter, useState } from '../../composables/vuex';
 import {
-  useAccounts,
   useMultisigAccounts,
-  usePendingMultisigTransaction,
   useTransactionAndTokenFilter,
-  useTransactionTx,
+  useViewport,
+  useAccounts,
+  usePendingMultisigTransaction,
 } from '../../composables';
-
 import TransactionItem from './TransactionItem.vue';
 import AnimatedSpinner from '../../icons/animated-spinner.svg?skip-optimize';
-import Visible from '../../icons/visible.svg?vue-component';
 import InfiniteScroll from './InfiniteScroll.vue';
+import {
+  IDexContracts,
+  ITokenList,
+  ITransaction,
+  ITransactionElement,
+  ITransactionsState,
+  ITx,
+} from '../../types';
+import { i18n } from '../../store/plugins/languages';
 
 export default defineComponent({
   components: {
     InfiniteScroll,
     TransactionItem,
     AnimatedSpinner,
-    Visible,
   },
   props: {
     token: { type: String, default: '' },
-    maxLength: { type: Number, default: null },
-    scrollTopThreshold: { type: Number, default: undefined },
-    showFilters: Boolean,
-    showSearch: Boolean,
     isMultisig: Boolean,
   },
   setup(props, { root }) {
     const loading = ref(false);
     const isDestroyed = ref(false);
+    const fetchTransactions = useDispatch('fetchTransactions');
+    const availableTokens = useState<ITokenList>('fungibleTokens', 'availableTokens');
+    const transactions = useState<ITransactionsState>('transactions');
+    const getTxSymbol = useGetter('getTxSymbol');
+    const getTxDirection = useGetter('getTxDirection');
+    const getDexContracts = useGetter<IDexContracts>('getDexContracts');
 
-    const { activeAccount } = useAccounts({ store: root.$store });
-    const { activeMultisigAccount } = useMultisigAccounts({ store: root.$store });
+    const getAccountPendingTransactions = useGetter<ITransaction[]>('getAccountPendingTransactions');
 
-    const { pendingMultisigTransaction } = usePendingMultisigTransaction({ store: root.$store });
+    const canLoadMore = computed(() => !!transactions.value.nextPageUrl);
```

Comment:
Moving the getters and computed properties above the composables in my opinion is a bad thing. In most cases we the order of the logic in the setup function is:
1. Composables (use...)
2. Refs
3. Getters
4. Computed
5. Methods/Functions
6. Hooks